### PR TITLE
Refactor report

### PR DESCRIPTION
**Pull request by IshikaM19**
I changed the data source so now we're pulling from yesterdays data instead of all data.

### CODE REVIEW

File: report/report.py [hunk 1]
```diff
@@ -71,13 +42,13 @@ def load_yesterday_data(db_connection: extensions.connection) -> pd.DataFrame:
                     JOIN item_genre
                     ON item_genre.item_id = item.item_id
                     JOIN genre
-                    ON genre.genre_id = item_genre.genre_id
+                    ON genre.genre_id =item_genre.genre_id
                     JOIN item_type
                     ON item_type.item_type_id = item.item_type_id
-                    WHERE DATE(sale_time) = CURRENT_DATE - INTERVAL '1 day'""")
+                    WHERE DATE(sale_time) = CURRENT_DATE - INTERVAL '1 day';""")
```
Comment (caithomas6789) [inline report/report.py:51]: Good practice to remember the semi-colons of SQL queries.

File: report/report.py [hunk 2]
```diff
@@ -534,9 +505,9 @@ def handler(event=None, context=None) -> dict:
 
     connection = get_db_connection()
 
-    all_data = load_all_data(connection)
+    yesterday_data = load_yesterday_data(connection)
```
Comment (caithomas6789) [inline report/report.py:515]: I like how the script retrieves yesterday's data for the report.

**Review by caitlinturnidge — APPROVED** [review]

File: report/report.py [hunk 3]
```diff
@@ -547,4 +518,12 @@ def handler(event=None, context=None) -> dict:
...
-    print(handler())
+    load_dotenv()
+
+    connection = get_db_connection()
+
+    yesterday_data = load_yesterday_data(connection)
+
+    html_string = generate_html_string(yesterday_data)
+
+    convert_html_to_pdf(html_string, './Bandcamp-Daily-Report.pdf')
```
Comment (IB64) [inline report/report.py]: Very nice separation of code into functions! Makes sections like this look really nice

File: report/report.py [hunk 4]
```diff
@@ -512,39 +460,81 @@ def convert_html_to_pdf(source_html, output_filename):
...
-    # close output file
-    result_file.close()                 # close output file
+    result_file.close()
 
-    # return True on success and False on errors
     return pisa_status.err
 
 
-def handler(event=None, context=None) -> dict:
+def send_email(report_file_path: str):
```
Comment (caithomas6789) [inline report/report.py:475]: Good way to send an email instead of using SNS.

File: report/report.py [hunk 5]
```diff
@@ -54,38 +58,59 @@ def load_all_data(db_connection: extensions.connection) -> pd.DataFrame:
...
+        html_string += f"""
+        <tr>
+        <td>{item[f'{key}']}</td>
+        <td>{item[f'{value}']}</td>
+        <td>{item[f'{value_2}']}</td>
+        </tr>"""
+
+    html_string += "</table>"
+
+    return html_string
```
Comment (caithomas6789) [inline report/report.py:106]: Using these functions really cleans up the code later in the script.

File: report/report.py [hunk 6]
```diff
@@ -443,6 +387,7 @@ def generate_html_string(data: pd.DataFrame) -> str:
             </p>
         </div>
         <div class="new-page" class="footer">
+            <img class="header" src="./bandcamp_logo.jpeg"  style="width:200px;height:100px;">
```
Comment (caithomas6789) [inline report/report.py:390]: I like the inclusion of the bandcamp logo.

**Review by caithomas6789 — APPROVED** [review]

**Pull request merged**